### ISSUE TITLE
Actor future can't be passed to submit

### ISSUE THREAD

**Issue by YarShev** [dask/distributed#4488]
**What happened**:

Actor future can't be passed to submit.

**What you expected to happen**:

It is expected that actor future can be passed to submit with same way as regular future.

**Minimal Complete Verifiable Example**:

```python
from distributed.client import Client
c = Client()
future1 = c.submit(lambda: (1, 2))
futures = [c.submit(lambda l: l[i], future1) for i in range(2)]
futures[0].result()
1
futures[1].result()
2
class Actor:
    def foo(self):
        return (1, 2)
actor_future = c.submit(Actor, actor=True)
actor = actor_future.result()
future2 = actor.foo()
futures = [c.submit(lambda l: l[i], future2) for i in range(2)]
TypeError: cannot pickle '_thread.lock' object
```

**Anything else we need to know?**:

**Environment**:

- Dask version: 2.19.0
- Python version: Python 3.8.5
- Operating System: both Windows and Linux
- Install method (conda, pip, source): conda

**Comment by YarShev** [comment]
https://github.com/dask/dask/discussions/7115

**Comment by martindurant** [comment]
Noting from the discussion referenced above, that this may never be implemented, since `future2` does not point to a result held by any worker.

Why is your formulation preferable to
```
c.submit(lambda: actor.foo().result()[i])
```

**Comment by YarShev** [comment]
Yes, my example works by following your suggestion but it would be nice not to care of a blocking `result()` call as that is done in ray, for instance. Does no one seek to make Actors' functionality better in Dask?

**Comment by martindurant** [comment]
Yes, I do, but it is not a priority. To be sure, Dask is not Ray, they have different architectures and different focus on use cases.

> it would be nice not to care of a blocking result() call

This is happening in the worker; passing an ordinary future would also not allow the task to proceed until that future was ready, so I don't see much of a difference.

If you have an interest in pushing actors forward, a set of design patterns for their use would help us. Why actors, and how they would be used, why this is the best pattern.

**Comment by YarShev** [comment]
> This is happening in the worker; passing an ordinary future would also not allow the task to proceed until that future was ready, so I don't see much of a difference.

A user/developer has to take care of that. This is the main concern.



> If you have an interest in pushing actors forward, a set of design patterns for their use would help us. Why actors, and how they would be used, why this is the best pattern.

As a start point, I think, it would be enough that what ray actors can do.

**Comment by martindurant** [comment]
> it would be enough that what ray actors can do.

I'm sorry, but this is not very helpful. Attempting to put in a set of features without a clear idea of the use case is likely to be misplaced effort. You could, after all, use Ray already. What *specifically* are you trying to achieve, and why would actors be the way to do it?

Again, the main point here is, that methods called on an actor generally cause a change of state, so you should not expect subsequent calls to return the same. So if you were to pass actor futures around, each would need to know when it was called, and the set of possible results would need to be stored. This is no small undertaking. You *could* more easily drop this requirement and accept that the same future may give different results at different times.

**Comment by YarShev** [comment]
My goal is to have Dask actor's functionality and if it was better/faster than Ray actors' functionality, I would use Dask for some purposes and Ray for other ones. One of the use cases is to have an actor that occupies a whole node and uses all cores to perform a function.

**Comment by martindurant** [comment]
I'm sorry, this is *not* a usecase. What's the computation? 

Dask already does a very good job at "occupying cores". Why do you want Dask to be like Ray? No matter what we did, we would be unlikely to match their actors' performance, since Ray was designed from the start with this pattern in mind.
